src/ast/passes/semantic_analyser.cpp

```diff
@@ -4021,6 +4040,31 @@ bool SemanticAnalyser::update_string_size(SizedType &type,
   return false;
 }
 
+SizedType SemanticAnalyser::create_merged_tuple(const SizedType &left,
+                                                const SizedType &right)
+{
+  assert(left.IsTupleTy() && right.IsTupleTy() &&
+         (left.GetFieldCount() == right.GetFieldCount()));
+
+  SizedType type;
+  type.SetSize(std::max(left.GetSize(), right.GetSize()));
```

Comment:
This is IMO incorrect. You can have two tuples like `((int64)0, (int32)1)` and `((int32)1, (int64)0)` which will both have size 12 but the merged tuple should have the size 16 (`(int64, int64)`).

Reply:
Good catch. I think I can get rid of this line entirely as it gets ignored anyway for the case above, as the tuple gets resized based on it's elements anyway.